src/NServiceBus.PersistenceTests/Sagas/When_using_storage_sessions_concurrently.cs

```diff
@@ -0,0 +1,80 @@
+namespace NServiceBus.PersistenceTesting.Sagas;
+
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+public class When_using_storage_sessions_concurrently(TestVariant param) : SagaPersisterTests(param)
+{
+    [Test]
+    public async Task It_should_dispose_correctly()
+    {
+        // 1 : Open
+        // 1 : Commit
+        // 2 : Open
+        // 1 : Dispose
+        // 2 : Commit
+
+        var correlationPropertyData = Guid.NewGuid().ToString();
+        var sagaData = new TestSagaData { SomeId = correlationPropertyData, DateTimeProperty = DateTime.UtcNow };
+        await SaveSaga(sagaData);
+        var generatedSagaId = sagaData.Id;
+
+        var persister = configuration.SagaStorage;
+
+        //  1: Open
+        var session1Context = configuration.GetContextBagForSagaStorage();
+        using var session1StorageSession = configuration.CreateStorageSession();
+        await session1StorageSession.Open(session1Context);
+
+        var record = await persister.Get<TestSagaData>(generatedSagaId, session1StorageSession, session1Context);
+
+        record.DateTimeProperty = DateTime.UtcNow;
+        await persister.Update(record, session1StorageSession, session1Context);
+        // 1:  Commit
+        await session1StorageSession.CompleteAsync();
```

Comment:
We need the using in this case to have the block {} identifiers so that the initial dispose gets called before it gets called the second time.